includes/class-newspack-popups.php

```diff
@@ -1300,5 +1302,97 @@ public static function disable_prompts_for_protected_pages() {
 			}
 		}
 	}
+
+	/**
+	 * Migrate user data from the 'wp_newspack_campaigns_reader_events' table to
+	 * the reader data library.
+	 */
+	public static function migrate_user_data() {
+		$option_name    = 'newspack_popups_reader_data_migrated';
+		$user_meta_name = 'newspack_popups_reader_data_migrated';
+
+		if (
+			get_option( $option_name ) ||
+			! is_user_logged_in() ||
```

Comment:
On a second thought, that will be caling plugin's `Reader_Activation::is_user_reader()` on sites that are not necessarily RAS. Or we can make a check if the roles are `subscriber` or `customer`, but perhaps the site runs custom roles through custom plugins.

I think I'd rather migrate everyone.

Reply:
Where are you creating this option? 

I suggest cleaning the cache after running the `update_option` call to avoid that bug we're seeing in persistent cache

Reply:
> Where are you creating this option?

To not run this after the migration is done

Reply:
I mean where not why. I didn't find it in the diff

Reply:
It was before it was removed here: fe77e90b867a99da2bd1d82e84530303b6cae936